syntax = "proto3";

package common;

option java_package = "com.lsorter.common";
option java_outer_classname = "CommonMessagesProto";

message BoundingBox {
  int32 ymin = 1;
  int32 xmin = 2;
  int32 ymax = 3;
  int32 xmax = 4;
  string label = 5;
  float score = 6;
}

message ListOfBoundingBoxes {
  repeated BoundingBox packet = 1;
}

message BoundingBoxWithIndex {
  int32 index = 1;
  BoundingBox bb = 2;
}

message ListOfBoundingBoxesWithIndexes {
  repeated BoundingBoxWithIndex packet = 1;
}

message ImageRequest {
  bytes image = 1;
  int32 rotation = 2;
}

message Empty {

}

message SorterConfiguration {
  int32 speed = 1;
}

message SorterConfigurationWithIP {
  int32 speed = 1;
  string deviceIP = 2;
}

message ImageStore {
  bytes image = 1;
  int32 rotation = 2;
  string label = 3;
}